### PR TITLE
Add decompression tracing instrumentation.

### PR DESCRIPTION
**Pull request by cyriltovena**
This adds a collection of decompression statistics in the logql engine.

/cc @Dieterbe I think this could be the be propagated from there to the API.

Signed-off-by: Cyril Tovena <cyril.tovena@gmail.com>

### CODE REVIEW

**Comment by cyriltovena** [comment]
hold on I want to add deduping time.

File: pkg/chunkenc/memchunk.go [hunk 1]
```diff
@@ -669,19 +682,24 @@ func (si *bufferedIterator) Error() error { return si.err }
...
-		}
-		si.origBytes = nil
-		si.bufReader = nil
-		si.buf = nil
-		si.decBuf = nil
-		si.reader = nil
+		si.close()
 		return si.err
 	}
 	return si.err
```
Comment (rfratto) [inline pkg/chunkenc/memchunk.go]: It seems like we don't need `return si.err` twice now

Comment (cyriltovena) [reply pkg/chunkenc/memchunk.go]: true, I'm also going to improve so we can add more stats easily.

**Pull request merged**